src/main/sensors/gyro.c

```diff
@@ -942,6 +946,39 @@ static void checkForOverflow(gyroSensor_t *gyroSensor, timeUs_t currentTimeUs)
 #endif // USE_GYRO_OVERFLOW_CHECK
 }
 
+static void checkForYawSpin(gyroSensor_t *gyroSensor, timeUs_t currentTimeUs)
+{
+#ifdef USE_GYRO_YAW_SPIN_CHECK
+    // check for overflow to handle Yaw Spin on MPU gyros
+    if (gyroSensor->yawSpinDetected) {
+//        const float gyroOverflowResetRate = GYRO_OVERFLOW_RESET_THRESHOLD * gyroSensor->gyroDev.scale;
+        const float gyroYawSpinResetRate = gyroConfig()->gyro_yaw_spin_threshold * 14.0f;
```

Comment:
I don't understand this.  What is the 14.0 factor mean?  `gyro.gyroADCf[Z]` contains a value in degrees/sec so I'm not sure what the multiplication is for.  It looks like you going to try and compare to 14000dps based on your default value.

Probably need to come up with a better way to set the reset threshold since you're allowing the user to control the trigger threshold.  Possibly a percentage *might* be okay but has the negative effect of changing the window size based on the trigger.

Reply:
Also should consider combining the overflow and yaw spin detection functions.  In general the yaw spin is a subset of overflow but with different thresholds and only one axis.  The gyro loop is speed critical so making two calls for similar processing is not desirable.

Reply:
As to combining the two, yes that would be highly desirable.  

One code works for MPU gyros, the other is needed (and this code won't be helpful, generally) for ICM gyros.

So the 'user' should, if they have an MPU gyro, disable the overflow check and use this.  So their gyro utilisation will be less than what it is with the gyro overflow detect on, since this is a bit simpler.

The ICM user could use both, but any reasonable spin and the overflow code will override the anti-spin code, making it useless, practically.

So in my mind the ideal situation would be that the gyro overflow code would never be activated unless an ICM gyro was present.  Regardless of the user interface.  Now that I know what causes the rare spin/flyaways with MPU gyros, and we both agree and know that it is *not* sign reversal, there is absolutely zero need for overflow detection for MPU gyros.  But racers will benefit from optionally enabling anti-spin code.

That's how I see it, anyway.